### PR TITLE
Bump GLib to 2.68.0 (but also some other things, because why not)

### PR DESCRIPTION
**Pull request by ernestask**
This commit bumps the version of GLib to latest stable, which uses
Meson, so more changes were required to get it to build. The libffi bump
was necessary for
https://github.com/libffi/libffi/commit/ad0d1d239afc7ad09203847793dcc020856d1035,
without which linking would fail, as the library path is garbled.

Closes https://github.com/beaker-project/restraint/issues/9

### CODE REVIEW

**Comment by ernestask** [comment]
I got it to build, but then patching the pkg-config fails with `sed: can't read /root/restraint/third-party/tree/lib/pkgconfig/glib-2.0.pc: No such file or directory`, as `libdir` is lib64.

**Comment by StykMartin** [comment]
Can you please rebase @ernestask

**Comment by ernestask** [comment]
> Can you please rebase @ernestask

No.

**Comment by ernestask** [comment]
Rawhide seems to be https://mail.gnu.org/archive/html/bug-m4/2021-03/msg00000.html

**Comment by StykMartin** [comment]
> 
> 
> Rawhide seems to be https://mail.gnu.org/archive/html/bug-m4/2021-03/msg00000.html

Yes. That's known bug

**Comment by cbouchar** [comment]
> > Rawhide seems to be https://mail.gnu.org/archive/html/bug-m4/2021-03/msg00000.html
> 
> Yes. That's known bug

I have a patch for m4 but this other glib thing got in the way.

File: third-party/Makefile
Comment (cbouchar) [inline third-party/Makefile:237]: Should we remove these patch files too?

File: third-party/Makefile
Comment (cbouchar) [inline third-party/Makefile:239]: And remove this file as well as reference to it in restraint.spec.

**Comment by StykMartin** [comment]
Give me some rebase @ernestask

**Comment by TomasTomecek** [comment]
all green, what a blessed morning!

**Review by cbouchar — APPROVED** [review]
Changes look good to me.  Nice to wait for another review.

**Review by StykMartin — APPROVED** [review]
Well yeah. Changes are fine. Please do not merge this before the community meeting.
We will agree on the strategy tomorrow.

**Comment by StykMartin** [comment]
Thanks, @ernestask. This will help a lot.

**Pull request merged**
